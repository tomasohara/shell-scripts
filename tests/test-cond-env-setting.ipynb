{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a928559",
   "metadata": {},
   "outputs": [],
   "source": [
    "## SETUP (OPTIONAL - USE IF REQUIRED)\n",
    "## Bracketed Paste is disabled to prevent characters after output\n",
    "## Example: \n",
    "## $ echo 'Hii'\n",
    "## | Hi?2004l\n",
    "# bind 'set enable-bracketed-paste off'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "93ec4a3f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "## TEST:  Make sure simple prompt used (e.g., no escapes that might contaminate output)\n",
    "## PS1=\"$ \"\n",
    "## TODO: PS1=\"> \"\n",
    "\n",
    "## NOTE: The Jupyter bash kernel requires that PS1 not be modified as they customize it. \n",
    "# echo $PS1\n",
    "actual=${PS1: -1}\n",
    "[ \"$actual\" = \">\" ]; echo $?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c6ffbc59",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD (SUPER BAD)\n",
    "# # Delete all aliases and function\n",
    "# # TODO: Instead start from pristine environment\n",
    "# unalias -a\n",
    "# alias | wc -l\n",
    "# for f in $(typeset -f | egrep '^\\w+'); do unset -f $f; done\n",
    "# typeset -f | egrep '^\\w+' | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "452a278e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "alias testuser=\"sed -r \"s/\"$USER\"+/userxf333/g\"\"\n",
    "## For number of digits not being critical\n",
    "alias testnum=\"sed -r \"s/[0-9][0-9]*/N/g\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "233f11ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setting a temp directory for tests\n",
    "## OLD: TMP=/tmp/test-cond-env\n",
    "\n",
    "# OLD\n",
    "# ## NOTE: Source it directly from the ./tests directory.\n",
    "# BIN_DIR=$PWD/..\n",
    "# ## You will need to run jupyter from that directory.\n",
    "# ## source $TEMP_BIN/_dir-aliases.bash\n",
    "# source _dir-aliases.bash\n",
    "# alias | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "be943c71",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed '/tmp/test-cond-env-setting/cond-env-settings' -> '/tmp/test-cond-env-setting/cond-env-settings.01May24.7'\n"
     ]
    }
   ],
   "source": [
    "## NOTE: For reproducability, the directory name needs to be fixed\n",
    "## In place of $$, use a psuedo random number (e,g., 1710)\n",
    "## *** All output from one run to the next needs to be the same ***\n",
    "\n",
    "# # temp_dir=$TMP/test-$$\n",
    "# temp_dir=$TMP/test-1710\n",
    "# trash_dir=$TMP/\"_temp-trash-$(date \"+%Y%m%d%H%M%S\")\"\n",
    "# ## OLD\n",
    "# # mkdir -p \"$temp_dir\"\n",
    "# # # TODO: /bin/rm -rvf \"$temp_dir\"\n",
    "# # cd \"$temp_dir\"\n",
    "# command mkdir -p \"$temp_dir\"\n",
    "# command mkdir -p \"$trash_dir\"\n",
    "# # # TODO: /bin/rm -rvf \"$temp_dir\"\n",
    "# command cd \"$temp_dir\"\n",
    "\n",
    "TMP=${TMP:-/tmp}\n",
    "if [ \"$DEBUG_LEVEL\" -lt 4 ]; then TMP=/tmp/test-cond-env-setting; fi\n",
    "\n",
    "temp_dir=$TMP/cond-env-settings\n",
    "rename-with-file-date \"$temp_dir\"\n",
    "command mkdir -p \"$temp_dir\"\n",
    "command cd $temp_dir\n",
    "\n",
    "## OLD: For debugging\n",
    "# pwd\n",
    "\n",
    "## OLD\n",
    "## ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)\n",
    "# alias linebr=\"printf '%*s\\n' \"${COLUMNS:-$(tput cols)}\" '' | tr ' ' -\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a67f36df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "# Count aliases proper\n",
    "## OLD: alias | wc -l | testnum\n",
    "[ \"$(alias | wc -l)\" -gt 440 ]; echo $?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c39f70a7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "# Count functions\n",
    "## OLD: typeset -f | egrep '^\\w+' | wc -l | testnum\n",
    "[ \"$(typeset -f | egrep '^\\w+' | wc -l)\" -gt 480 ]; echo $?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e54c2879",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD\n",
    "# source $BIN_DIR/tomohara-aliases.bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bb9f1941",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "# SETTING A ENVIRONMENT VARIABLE NAMED COND_ENV_TEST_TEMP\n",
    "\n",
    "cond-setenv COND_ENV_TEST_TEMP tmp/cond-env-test-temp\n",
    "[ \"$COND_ENV_TEST_TEMP\" == \"tmp/cond-env-test-temp\" ]; echo $?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a59bf7fd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "# USE OF ENV VARIABLE TO CREATE A SUB-DIRECTORY\n",
    "# BAD: rm -rf ./* > /dev/null\n",
    "mkdir -p $COND_ENV_TEST_TEMP\n",
    "# OLD: ls -l | awk '!($6=$7=$8=\"\")' | testnum | testuser\n",
    "[[ \"$(ls | grep -c tmp)\" -eq 1 && \"$(ls tmp | grep -c cond-env-test-temp)\" -eq 1 ]]; echo $?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3d1256f7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "# THE OUTPUT OF printenv IS STORED IN temp-env-1.tmp CREATED IN THE DIRECTORY SPECIFIED BY $COND_ENV_TEST_TEMP\n",
    "## OLD: For debug\n",
    "# pwd\n",
    "printenv > $COND_ENV_TEST_TEMP/temp-env-1.tmp\n",
    "# cat $COND_ENV_TEST_TEMP/temp-env-1.tmp | grep -c \"shell-scripts\"\n",
    "[ \"$(cat $COND_ENV_TEST_TEMP/temp-env-1.tmp | grep -c \"shell-scripts\")\" ]; echo $?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3bb906ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "## OLD: Reworked and moved the validation to above cell\n",
    "# cat $COND_ENV_TEST_TEMP/temp-env-1.tmp | grep -q \"shell-scripts\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "4f67b1aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n"
     ]
    }
   ],
   "source": [
    "## OLD: No use og trash_dir: command mv -f * \"$trash_dir\"\n",
    "echo \"Done!\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
