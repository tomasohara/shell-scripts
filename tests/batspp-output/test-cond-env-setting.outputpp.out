1..14
not ok 1 test-1
# (in test file batspp-output/test-cond-env-setting.outputpp, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== echo $PS1
#  ==========
#
# ========= [PEXP\[\]ECT_PROMPT> =========
# [PEXP\[\]ECT_PROMPT>
# ============================
not ok 2 test-3
# (in test file batspp-output/test-cond-env-setting.outputpp, line 59)
#   `[ "$(test-3-actual)" == "$(test-3-expected)" ]' failed
# ========== unalias -a
#  ==========
# ========= $ alias | wc -l\n$ for f in ; do unset -f ; done\n$ typeset -f | egrep '^\w+' | wc -l\n0\n0 =========
# $ alias | wc -l
# $ for f in bats_abort_timeout_countdown ()
# bats_add_debug_exclude_path ()
# bats_all_in ()
# bats_any_in ()
# bats_append_arrays_as_args ()
# bats_approx_realpath ()
# bats_binary_search ()
# bats_capture_stack_trace ()
# bats_check_status_from_trap ()
# bats_check_valid_version ()
# bats_create_test_tmpdirs ()
# bats_debug_trap ()
# bats_emit_trace ()
# bats_error_trap ()
# bats_evaluate_preprocessed_source ()
# bats_execute ()
# bats_exit_trap ()
# bats_extract_line ()
# bats_format_file_line_reference ()
# bats_format_file_line_reference_colon ()
# bats_format_file_line_reference_comma_line ()
# bats_format_file_line_reference_uri ()
# bats_frame_filename ()
# bats_frame_function ()
# bats_frame_lineno ()
# bats_generate_warning ()
# bats_get_child_processes_of ()
# bats_get_failure_stack_trace ()
# bats_internal_load ()
# bats_interrupt_trap ()
# bats_interrupt_trap_in_run ()
# bats_kill_childprocesses_of ()
# bats_load_library ()
# bats_load_library_safe ()
# bats_load_safe ()
# bats_merge_stdout_and_stderr ()
# bats_normalize_windows_dir_path ()
# bats_perform_test ()
# bats_prefix_lines_for_tap_output ()
# bats_preprocess_source ()
# bats_print_failed_command ()
# bats_print_stack_trace ()
# bats_quote_code ()
# bats_readlinkf ()
# declare -fx bats_readlinkf
# bats_redirect_stderr_into_file ()
# bats_replace_filename ()
# bats_require_minimum_version ()
# bats_separate_lines ()
# bats_setup_tracing ()
# bats_should_retry_test ()
# bats_sort ()
# bats_start_timeout_countdown ()
# bats_strip_string ()
# bats_teardown_trap ()
# bats_test_begin ()
# bats_test_function ()
# bats_timeout_trap ()
# bats_trim ()
# bats_trim_filename ()
# bats_version_lt ()
# bats_warn_minimum_guaranteed_version ()
# find_in_bats_lib_path ()
# get_mills_since_epoch ()
# load ()
# run ()
# setup ()
# skip ()
# teardown ()
# test-1-actual ()
# test-1-expected ()
# END_EXPECTED
# test-10-actual ()
# test-10-expected ()
# END_EXPECTED
# test-11-actual ()
# test-11-expected ()
# tmp/cond-env-test-temp
# END_EXPECTED
# test-12-actual ()
# test-12-expected ()
# total 4
# drwxrwxr-x 3 userxf333 userxf333 4096    tmp
# END_EXPECTED
# test-13-actual ()
# test-13-expected ()
# END_EXPECTED
# test-14-actual ()
# test-14-expected ()
# TOM_BIN=/home/userxf333/bin
# END_EXPECTED
# test-15-actual ()
# test-15-expected ()
# Done!
# END_EXPECTED
# test-3-actual ()
# test-3-expected ()
# 0
# 0
# END_EXPECTED
# test-4-actual ()
# test-4-expected ()
# END_EXPECTED
# test-5-actual ()
# test-5-expected ()
# 9
# END_EXPECTED
# test-6-actual ()
# test-6-expected ()
# END_EXPECTED
# test-7-actual ()
# test-7-expected ()
# 10
# END_EXPECTED
# test-8-actual ()
# test-8-expected ()
# 2
# END_EXPECTED
# test-9-actual ()
# test-9-expected ()
# END_EXPECTED
# test_test-2d1 ()
# test_test-2d10 ()
# test_test-2d11 ()
# test_test-2d12 ()
# total 4
# drwxrwxr-x 3 userxf333 userxf333 4096    tmp' "=========";
# test_test-2d13 ()
# test_test-2d14 ()
# test_test-2d15 ()
# test_test-2d3 ()
# test_test-2d4 ()
# test_test-2d5 ()
# 9' "=========";
# test_test-2d6 ()
# test_test-2d7 ()
# test_test-2d8 ()
# test_test-2d9 () ; do unset -f ; done
# $ typeset -f | egrep '^\w+' | wc -l
# 0
# 0
# ============================
ok 3 test-4
not ok 4 test-5
# (in test file batspp-output/test-cond-env-setting.outputpp, line 131)
#   `[ "$(test-5-actual)" == "$(test-5-expected)" ]' failed
# ========== TMP=/tmp/test-cond-env
#  ==========
# ========= ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=$PWD/..
# ## You will need to run jupyter from that directory.
# ## source $TEMP_BIN/_dir-aliases.bash
# $ source _dir-aliases.bash
# $ alias | wc -l
# 9 =========
# ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=/tmp/batspp-76414/test-5/..
# ## You will need to run jupyter from that directory.
# ## source /_dir-aliases.bash
# $ source _dir-aliases.bash
# $ alias | wc -l
# 9
# ============================
not ok 5 test-6
# (in test file batspp-output/test-cond-env-setting.outputpp, line 171)
#   `[ "$(test-6-actual)" == "$(test-6-expected)" ]' failed
# ========== temp_dir=$TMP/test-1710
#  ==========
# ========= $ mkdir -p "$temp_dir"
# # TODO: /bin/rm -rvf "$temp_dir"
# $ cd "$temp_dir"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -"
# /tmp/test-cond-env/test-1710 =========
# $ mkdir -p "/test-1710"
# # TODO: /bin/rm -rvf "/test-1710"
# $ cd "/test-1710"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "80" '' | tr ' ' -"
# /tmp/test-cond-env/test-1710
# ============================
not ok 6 test-7
# (in test file batspp-output/test-cond-env-setting.outputpp, line 211)
#   `[ "$(test-7-actual)" == "$(test-7-expected)" ]' failed
# ========== alias | wc -l
#  ==========
# 0
# ========= 10 =========
# 10
# ============================
not ok 7 test-8
# (in test file batspp-output/test-cond-env-setting.outputpp, line 245)
#   `[ "$(test-8-actual)" == "$(test-8-expected)" ]' failed
# ========== typeset -f | egrep '^\w+' | wc -l\n ==========
# 138
# ========= 2 =========
# 2
# ============================
not ok 8 test-9
# (from function `test-9-actual' in file batspp-output/test-cond-env-setting.outputpp, line 286,
#  in test file batspp-output/test-cond-env-setting.outputpp, line 273)
#   `test-9-actual ' failed
# ========== source $BIN_DIR/tomohara-aliases.bash
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/test-cond-env-setting.outputpp: line 286: /tomohara-aliases.bash: No such file or directory
not ok 9 test-10
# (from function `test-10-actual' in file batspp-output/test-cond-env-setting.outputpp, line 320,
#  in test file batspp-output/test-cond-env-setting.outputpp, line 307)
#   `test-10-actual ' failed with status 127
# ========== cond-setenv COND_ENV_TEST_TEMP tmp/cond-env-test-temp
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/test-cond-env-setting.outputpp: line 320: cond-setenv: command not found
not ok 10 test-11
# (in test file batspp-output/test-cond-env-setting.outputpp, line 347)
#   `[ "$(test-11-actual)" == "$(test-11-expected)" ]' failed
# ========== echo $COND_ENV_TEST_TEMP
#  ==========
#
# ========= tmp/cond-env-test-temp =========
# tmp/cond-env-test-temp
# ============================
not ok 11 test-12
# (in test file batspp-output/test-cond-env-setting.outputpp, line 381)
#   `[ "$(test-12-actual)" == "$(test-12-expected)" ]' failed
# ========== rm -rf ./* > /dev/null
#  ==========
# ========= $ mkdir -p $COND_ENV_TEST_TEMP
# $ ls -l | awk '!($6=$7=$8="")' | testuser
# total 4
# drwxrwxr-x 3 userxf333 userxf333 4096    tmp =========
# $ mkdir -p
# $ ls -l | awk '!(==="")' | testuser
# total 4
# drwxrwxr-x 3 userxf333 userxf333 4096    tmp
# ============================
not ok 12 test-13
# (in test file batspp-output/test-cond-env-setting.outputpp, line 418)
#   `[ "$(test-13-actual)" == "$(test-13-expected)" ]' failed
# ========== pwd
#  ==========
# /tmp/batspp-76414/test-13
# ========= $ printenv >> $COND_ENV_TEST_TEMP/temp-env-1.tmp
# /tmp/test-cond-env/test-1710 =========
# $ printenv >> /temp-env-1.tmp
# /tmp/test-cond-env/test-1710
# ============================
not ok 13 test-14
# (from function `test-14-actual' in file batspp-output/test-cond-env-setting.outputpp, line 460,
#  in test file batspp-output/test-cond-env-setting.outputpp, line 447)
#   `test-14-actual ' failed with status 127
# ========== cat $COND_ENV_TEST_TEMP/temp-env-1.tmp | grep TOM_BIN | testuser
#  ==========
# cat: /temp-env-1.tmp/home/aveey/tom-project/shell-scripts/tests/batspp-output/test-cond-env-setting.outputpp: line 460: testuser: command not found
# : No such file or directory
ok 14 test-15
